### PR TITLE
(Fix #185) Enhance TLS

### LINKED ISSUE #185: TLS: add a few extra settings
In addition to the `ssl-ciphers` setting I'd like to request that an `ssl-protocols` setting be added, so that for example protocols such as SSLv2 and SSLv3 can be disabled or only TLSv1.2 be allowed.

Maybe also a setting to disable compression? Or even hardcode it (probably the better option).

Also, since eggdrop acts as a server for its listening ports, a setting like `ssl-dhparam` for specifying a custom Diffie-Hellman parameters file is probably in order as well. https://weakdh.org/

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: Robby-
Patch by: michaelortmann
Fixes: #185

One-line summary:
Enhance TLS

Additional description (if needed):
1. Add new config variable ssl-protocols
2. Add new config variable ssl-dhparam
3. Drop insecure ssl compression
4. Add comment to eggdrop.conf that setting ssl-ciphers may not impact TLSv1.3 ciphersuites

Test cases demonstrating functionality (if applicable):
i tested the new code and variables with temp debug printf and with .link sslbot.
looks good.

### CODE REVIEW

**Comment by Cizzle** [comment]
Don't see why this couldn't go in 1.8.4; still lacking some docs though.

File: src/tls.c [hunk 1]
```diff
@@ -168,6 +170,94 @@ int ssl_init()
            ERR_error_string(ERR_get_error(), NULL));
     ERR_free_strings();
   }
+  /* Let advanced users specify the list of allowed ssl protocols */
+  #define EGG_SSLv2   1 << 0
```
Comment (thommey) [inline src/tls.c]: paranthesis around definition please (1 << 0), in case it's moved and used in odd places

**Review by thommey — DISMISSED** [review]

File: src/tls.c (hunk 1 above)
Comment (michaelortmann) [reply src/tls.c]: done

**Comment by michaelortmann** [comment]
@thommey: Requested changes done.

@Cizzle: Im no native speaker, but if you can tell me, **where** i should add documentation i will do.

**Comment by Cizzle** [comment]
At `help/core.help` (adding to the `set` help) and `help/set/cmds1.help`.

**Comment by michaelortmann** [comment]
@Cizzle: as far as i understand: `help/core.help` is more about "on the fly changeable" settings; the new settings are STR_PROTECT; so i think were done here. I added help text to `help/set/cmds1.help` as requested. eggdrop.conf was changed accordingly.

**Review by vanosg — APPROVED** [review]

**Review by thommey — APPROVED** [review]
This is fine but let's hope we don't get TLSv1.4 soon and have to redo this

**Pull request merged**